### PR TITLE
new resource `azurerm_spring_cloud_app_mysql_association`

### PR DESCRIPTION
**Pull request by njuCZ**
second PR to associate spring cloud app with mysql server

image

### CODE REVIEW

File: website/docs/r/spring_cloud_app_mysql_association.html.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* `id` - The ID of the Spring Cloud Application Mysql Association.
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
+
+* `create` - (Defaults to 30 minutes) Used when creating the Spring Cloud Application Mysql Association.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Spring Cloud Application Mysql Association.
+* `update` - (Defaults to 30 minutes) Used when updating the Spring Cloud Application Mysql Association.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Spring Cloud Application Mysql Association.
```
Comment (tombuildsstuff) [inline website/docs/r/spring_cloud_app_mysql_association.html.markdown:100]:
```suggestion
* `id` - The ID of the Spring Cloud Application MySQL Association.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Spring Cloud Application MySQL Association.
* `read` - (Defaults to 5 minutes) Used when retrieving the Spring Cloud Application MySQL Association.
* `update` - (Defaults to 30 minutes) Used when updating the Spring Cloud Application MySQL Association.
* `delete` - (Defaults to 30 minutes) Used when deleting the Spring Cloud Application MySQL Association.
```

File: website/docs/r/spring_cloud_app_mysql_association.html.markdown [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+## Import
+
+Spring Cloud Application Mysql Association can be imported using the `resource id`, e.g.
```
Comment (tombuildsstuff) [inline website/docs/r/spring_cloud_app_mysql_association.html.markdown:104]:
```suggestion
Spring Cloud Application MySQL Association can be imported using the `resource id`, e.g.
```

File: website/docs/r/spring_cloud_app_mysql_association.html.markdown [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+subcategory: "Spring Cloud"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_spring_cloud_app_mysql_association"
+description: |-
+  Associates a [Spring Cloud Application](spring_cloud_app.html) with a [Mysql Database](mysql_database.html).
```
Comment (tombuildsstuff) [inline website/docs/r/spring_cloud_app_mysql_association.html.markdown:6]:
```suggestion
  Associates a [Spring Cloud Application](spring_cloud_app.html) with a [MySQL Database](mysql_database.html).
```

File: website/docs/r/spring_cloud_app_mysql_association.html.markdown [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+* `username` - (Required) Specifies the username when connecting to Mysql Database.
+
+* `password` - (Required) Specifies the password when connecting to Mysql Database.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the Spring Cloud Application Mysql Association.
```
Comment (tombuildsstuff) [inline website/docs/r/spring_cloud_app_mysql_association.html.markdown:91]:
```suggestion
* `id` - The ID of the Spring Cloud Application MySQL Association.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @njuCZ 

Thanks for this PR.

I've taken a look through and left some comments inline but this is off to a good start - if we can fix up the comments then we should be able to take another look.

Thanks!

File: azurerm/internal/services/springcloud/spring_cloud_app_mysql_association_resource.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"username": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"password": {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/springcloud/spring_cloud_app_mysql_association_resource.go:79]: also this is a password, so isn't this a sensitive value?

**Comment by njuCZ** [comment]
@tombuildsstuff thanks for your review. I have updated this PR, all acctest could pass

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 👍

**Pull request merged**